src/Data/Text/Internal/Transformation.hs

```diff
@@ -0,0 +1,267 @@
+{-# LANGUAGE BangPatterns, CPP, MagicHash, Rank2Types, UnboxedTuples, TypeFamilies #-}
+{-# LANGUAGE TemplateHaskellQuotes #-}
+{-# LANGUAGE Trustworthy #-}
+{-# LANGUAGE UnliftedFFITypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE PartialTypeSignatures #-}
```

Comment:
You can remove at least `Rank2Types` (BTW its new name is `RanNTypes`), `TypeFamilies`, `TemplateHaskellQuotes`.